pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -37,131 +40,179 @@ func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
+				Name:        "mirror-static-pod-name",
+				Namespace:   metav1.NamespaceDefault,
+				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
 			},
+		}
+		normalPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("normal-pod-uid"),
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
+		}
+	)
+
+	testCase := []struct {
+		name         string
+		podList      []*v1.Pod
+		wantPod      *v1.Pod
+		expectPods   []*v1.Pod
+		expectGetPod *v1.Pod
+		expectUID    types.UID
+	}{
+		{
+			name:         "Get normal pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      normalPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: normalPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
+		{
+			name:         "Get static pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      staticPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: staticPod,
+			expectUID:    types.UID("static-pod-uid"),
 		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	// Tests that all regular pods are recorded correctly.
-	actualPods := podManager.GetPods()
-	if len(actualPods) != len(expectedPods) {
-		t.Errorf("expected %d pods, got %d pods; expected pods %#v, got pods %#v", len(expectedPods), len(actualPods),
-			expectedPods, actualPods)
 	}
-	for _, expected := range expectedPods {
-		found := false
-		for _, actual := range actualPods {
-			if actual.UID == expected.UID {
-				if !reflect.DeepEqual(&expected, &actual) {
-					t.Errorf("pod was recorded incorrectly. expect: %#v, got: %#v", expected, actual)
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+
+			// Tests that all regular pods, except for mirror pods, are recorded correctly.
+
+			sortOpt := cmpopts.SortSlices(func(a, b *v1.Pod) bool { return a.Name < b.Name })
+			if diff := cmp.Diff(actualPods, test.expectPods, sortOpt); diff != "" {
+				t.Errorf("actualPods and expectPods are not equal: %s", diff)
+			}
+
+			// Tests UID translation works as expected. Convert static pod UID for comparison only.
+			if uid := podManager.TranslatePodUID(mirrorPod.UID); uid != kubetypes.ResolvedPodUID(test.expectUID) {
+				t.Errorf("unable to translate UID %q to the static POD's UID %q; %#v",
+					mirrorPod.UID, staticPod.UID, podManager.mirrorPodByUID)
+			}
+			fullName := fmt.Sprintf("%s_%s", test.wantPod.Name, test.wantPod.Namespace)
+
+			// Test the basic Get methods.
+			actualPod, ok := podManager.GetPodByFullName(fullName)
+			if !ok {
+				if diff := cmp.Diff(actualPod, test.expectGetPod); diff != "" {
+					t.Errorf("unable to get pod by full name; got: %#v, expected: %#v", actualPod, test.expectGetPod)
```

Comment:
Why not print `diff` here? Same for l.128.

```suggestion
					t.Errorf("unable to get pod by full name: %s", diff)
```

Reply:
Thanks for the reminder, fixed